### PR TITLE
195 implement metatranscriptomics workflow definition in workflow automation framework

### PR DESCRIPTION
**Pull request by mbthornton-lbl**
This PR provides updates to test fixtures and unit tests to enable the MetaTranscriptomics worflow configuration to be run in the workflow automation framework.

**Updates to test fixtures:**
- [x] bring existing test fixtures into line with current schema
- [x] add test fixtures for metaT 
- [x] DRY up and refactor existing test fixtures
- [x] Add fixtures/db_utils.py for test_db setup / teardown and fixture loading

**Updated Tests to cover workflow automation functionality with data fixtures for both MetaG and MetaT**

**Scheduler** tests for both metaG and metaT workflows
- [x] test_submit
- [x] test_progress

**Load Activities** tests for Job creation for MetaG and MetaT workflows
- [x] test_activities 

**WorflowJob Model**

**Workflow Utils**
- [x] Update Workflow model to include Analyte Category

**Other**
- move workflows.yaml files to be under nmdc_automation/config to fix path issues
- bring `test_data` into `tests/`

### CODE REVIEW

File: tests/fixtures/metagenome_assembly_set.json [hunk 1]
```diff
@@ -14,19 +14,20 @@
...
+    "git_url": "https://github.com/microbiomedata/metaAssembly",
+    "version": "v1.0.3",
     "has_output": [
-      "nmdc:37573bca240f88091720ae61ae5c9452",
-      "nmdc:220c85a9e2d470157aed7a7556f85656",
-      "nmdc:c3298d022868e8b0e7cfb783015b93ed",
-      "nmdc:3fc528f0105fc163876d68df26d35b29",
-      "nmdc:da186671c5a581af0a7d195bd857c871"
+      "nmdc:dobj-11-contigs",
+      "nmdc:dobj-11-cover-stats",
```
Comment (aclum) [inline tests/fixtures/metagenome_assembly_set.json:21]: I dont think the records with three dashes in them are going to be schema-valid. I would like to see integration tests where the example data is validated against the schema, either with the python package or linkml-validate

Comment (aclum) [reply tests/fixtures/metagenome_assembly_set.json:21]: errors from runtime json:validate.
{
  "result": "errors",
  "detail": {
    "metagenome_annotation_activity_set": [
      "'nmdc:dobj-11-proteins-faa' does not match '^(nmdc):dobj-([0-9][a-z]{0,6}[0-9])-([A-Za-z0-9]{1,})$'",
      "'nmdc:dobj-11-scaffold-lineage' does not match '^(nmdc):dobj-([0-9][a-z]{0,6}[0-9])-([A-Za-z0-9]{1,})$'",
      "'nmdc:dobj-11-ko-ec' does not match '^(nmdc):dobj-([0-9][a-z]{0,6}[0-9])-([A-Za-z0-9]{1,})$'",
      "'nmdc:dobj-11-product-names' does not match '^(nmdc):dobj-([0-9][a-z]{0,6}[0-9])-([A-Za-z0-9]{1,})$'",
      "'nmdc:dobj-11-gene-phylo' does not match '^(nmdc):dobj-([0-9][a-z]{0,6}[0-9])-([A-Za-z0-9]{1,})$'",
      "'nmdc:dobj-11-annotation-stats' does not match '^(nmdc):dobj-([0-9][a-z]{0,6}[0-9])-([A-Za-z0-9]{1,})$'",
      "'nmdc:dobj-11-contig-mapping' does not match '^(nmdc):dobj-([0-9][a-z]{0,6}[0-9])-([A-Za-z0-9]{1,})$'",
      "'nmdc:dobj-11-annotation-info' does not match '^(nmdc):dobj-([0-9][a-z]{0,6}[0-9])-([A-Za-z0-9]{1,})$'",
      "'nmdc:omprc-11-test-metag1' does not match '^(nmdc):(omprc)-([0-9][a-z]{0,6}[0-9])-([A-Za-z0-9]{1,})$'"
    ]
  }
}

File: tests/test_activities.py [hunk 2]
```diff
@@ -1,106 +1,102 @@
...
-
-    allowlist = set(["foo"])
-    acts = load_activities(db, wfs, allowlist=allowlist)
-    assert len(acts) == 0
-
-    allowlist = set(["nmdc:omprc-11-nhy4pz43"])
-    acts = load_activities(db, wfs, allowlist=allowlist)
-    assert len(acts) == 5
+    if metatranscriptome:
+        # seq, rqc, assembly, annotation, expression
```
Comment (aclum) [inline tests/test_activities.py:80]: not necessarily in this PR but seq needs to be deprecated.

File: tests/test_activities.py [hunk 3]
```diff
@@ -1,106 +1,102 @@
...
+        exp_num_acts = 5
+    else:
+        # seq, rqc, assembly, annotation, MAGs, read-based taxonomic annotation
```
Comment (aclum) [inline tests/test_activities.py:83]: same here not necessarily, deprecated seq at some point.

File: tests/test_sched.py [hunk 4]
```diff
@@ -1,36 +1,13 @@
...
-FIXTURE_DIR = Path(__file__).parent / "fixtures"
-
-
-@fixture
-def db():
-    conn_str = os.environ.get("MONGO_URL", "mongodb://localhost:27017")
-    return MongoClient(conn_str).test
-
+TRIGGER_SET = 'metagenome_annotation_activity_set'
+TRIGGER_ID = 'nmdc:55a79b5dd58771e28686665e3c3faa0c'
```
Comment (aclum) [inline tests/test_sched.py:9]: do these need to be updated to nmdc style IDs?

File: tests/fixtures/read_qc_analysis_activity_set2.json
Comment (aclum) [inline tests/fixtures/read_qc_analysis_activity_set2.json:1]: Consider setting these files up to all include what collection set they are part of, then we could use runtime to validate that they are schema valid.

File: nmdc_automation/config/config.py [hunk 5]
```diff
@@ -90,8 +96,10 @@ def client_secret(self):
     @property
     def allowed_workflows(self):
         """Generate a list of allowed workflows."""
-        workflows_config_path = self.config_data["workflows"]["workflows_config"]
-        with open(workflows_config_path, "r") as stream:
+        workflows_config_file = self.config_data["workflows"]["workflows_config"]
+        # print current working directory
+        print(os.getcwd())
```
Comment (scanon) [inline nmdc_automation/config/config.py:101]: Should probably drop this.

File: nmdc_automation/config/workflows/workflows.yaml [hunk 6]
```diff
@@ -314,7 +314,6 @@ Workflows:
       map_file: do:Contig Mapping File
       gene_phylogeny_file: do:Gene Phylogeny tsv
       proj: "{activity_id}"
-      map_file: do:Contig Mapping File
```
Comment (scanon) [inline nmdc_automation/config/workflows/workflows.yaml:317]: Why is this being removed?

Comment (aclum) [reply nmdc_automation/config/workflows/workflows.yaml:317]: its a duplicate. see line 314.

File: tests/fixtures/read_qc_analysis_activity_set2.json
Comment (aclum) [reply tests/fixtures/read_qc_analysis_activity_set2.json:1]: i think these are job ids which dont look like nmdc schema style IDs.

File: tests/test_activities.py (hunk 2 above)
Comment (mbthornton-lbl) [reply tests/test_activities.py]: My preference would be for another small feature branch - in the interest of getting this PR landed

File: tests/test_activities.py [hunk 7]
```diff
@@ -58,45 +24,110 @@ def test_activies(test_db, workflow_file, config_dir, fixtures_dir):
...
-    wfs = load_workflows(config_dir / "workflows.yaml")
-    assert wfs is not None
+    metatranscriptome = False
+    if workflow_file == "workflows-mt.yaml":
+        metatranscriptome = True
+
+    shared_wf_names = ["Sequencing Noninterleaved", "Sequencing Interleaved"]
+    if metatranscriptome:
+        exp_num_wfs = 9
+        exp_wf_names = ["Metatranscriptome Reads QC",
```
Comment (aclum) [inline tests/test_activities.py:68]: all of these should not run. It should be "Sequencing Noninterleaved" OR "Sequencing Interleaved", "Metatranscriptome Reads QC" OR  "Metatranscriptome Reads QC Interleave",  "Expression Analysis Antisense" OR 
                                      "Expression Analysis Sense" OR
                                      "Expression Analysis Nonstranded"
                                      
                                      exp_num_wfs = 5

Comment (aclum) [reply tests/test_activities.py:68]: also lets remove "Sequencing Noninterleaved" OR "Sequencing Interleaved", so this should be 4.

File: tests/test_activities.py [hunk 8]
```diff
@@ -58,45 +24,110 @@ def test_activies(test_db, workflow_file, config_dir, fixtures_dir):
...
+                                      "Metatranscriptome Reads QC Interleave",
+                                      "Metatranscriptome Assembly",
+                                      "Metatranscriptome Annotation",
+                                      "Expression Analysis Antisense",
+                                      "Expression Analysis Sense",
+                                      "Expression Analysis Nonstranded",
+                                      ]
+    else:
+        exp_num_wfs = 8
+        exp_wf_names = ["Reads QC", "Reads QC Interleave", "Metagenome Assembly",
```
Comment (aclum) [inline tests/test_activities.py:78]: It should be "Reads QC" OR "Reads QC Interleave"

**Review by aclum — CHANGES_REQUESTED** [review]
See comments on test_load_workflows. 

Also not a blocker but for all test data 'name' should be the file name, it is currently a mix of a description and a file name.  Also empty keys should be left out, there are several times in the test data where alternative_identifiers is present but empty.

File: tests/test_activities.py (hunk 7 above)
Comment (mbthornton-lbl) [reply tests/test_activities.py:68]: Added a docstring to load_workflows

**Comment by mbthornton-lbl** [comment]
Simplified `name` values in DataObject - i.e. not file names, are intended to make the fixtures more maintainable

**Review by aclum — APPROVED** [review]

**Pull request merged**